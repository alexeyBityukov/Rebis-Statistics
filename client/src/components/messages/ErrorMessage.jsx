import React from 'react';
import Typography from '@material-ui/core/Typography';
import { withStyles, createStyles } from '@material-ui/core';

const style = ({ palette, spacing }) => createStyles({
  error: {
    background: palette.error[50],
    padding: spacing.unit,
    border: `1px solid ${palette.error.main}`,
  },
});

export default withStyles(style)(({ classes, label }) => {
  if (label) {
    return (
      <Typography
        align="left"
        className={classes.error}
        color="error"
        variant="subtitle1"
      >
        {`* ${label}`}
      </Typography>
    );
  }
  return (null);
});
