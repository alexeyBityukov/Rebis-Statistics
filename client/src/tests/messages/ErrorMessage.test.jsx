import React from 'react';
import { configure, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import ErrorMessage from '../../components/messages/ErrorMessage';

configure({ adapter: new Adapter() });


describe('<ErrorMessage>', () => {
  it('should render correctly', () => {
    const component = shallow(
      <ErrorMessage label="Some error message" />,
    ).dive();
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('should nothing render', () => {
    const component = shallow(
      <ErrorMessage />,
    ).dive();
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
