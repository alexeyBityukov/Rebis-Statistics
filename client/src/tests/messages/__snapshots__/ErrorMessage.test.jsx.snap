// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage> should nothing render 1`] = `""`;

exports[`<ErrorMessage> should render correctly 1`] = `
<WithStyles(Typography)
  align="left"
  className="Component-error-1"
  color="error"
  variant="subtitle1"
>
  * Some error message
</WithStyles(Typography)>
`;
